### ISSUE TITLE
[ PackageControl | Fetch ] Thread Exception / KeyError 'load_order'

### ISSUE THREAD

**Issue by Enteleform** [wbond/package_control#1104]
**ST Version:** Stable Channel, Build 3103
**OS:** Windows 10 Pro (64-bit), Build 10586

&nbsp;
I'm not really having any issues or anything, ST seems to be working fine.

Just figured I'd post this for reference.
&nbsp;

This is the `console` output after ST is first opened:
```
DPI scale: 1
startup, version: 3103 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Users/Fico/OneDrive/( Scripts )/AutoHotKey
packages path: /C/Users/Fico/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/Fico/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/Fico/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["_DEV_SublimeStarterQuickPanel", "_MOD_VintageLines", "Alignment", "ASP", "BracketHighlighter", "Clickable URLs", "Clipboard Manager", "ClipboardManager", "Clojure", "Color Highlighter", "ColorPicker", "Comment-Snippets", "CursorRuler", "D", "Diff", "EasyDiff", "Erlang", "FileDiffs", "Fold Python", "Go", "Graphviz", "Groovy", "Haskell", "Highlighter", "LaTeX", "LineDiff", "Lisp", "MacroScript", "Makefile", "MarkdownHighlighting", "Matlab", "MaxPane", "Modific", "MultiEditUtils", "My Snippets", "OCaml", "Pascal", "Perl", "PlainTasks", "R", "RestructuredText", "Scala", "SFTP", "ShellScript", "SlideNav", "sublime-text-2-buildview", "SublimeLinter", "SublimeREPL", "SyncedSideBar", "SyncedSidebarBg", "SyntaxFold", "TCL", "Textile", "TrailingSpaces", "VAlign", "Vintage", "z__SublimeStarter_QuickPanel"]
pre session restore time: 0.314618
startup time: 0.379618
first paint time: 0.427618
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.profile_syntax_definition
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin Python.syntax_test_python
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.02-bz2
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin AlignTab.aligner
reloading plugin AlignTab.aligntab
reloading plugin AlignTab.hist
reloading plugin AlignTab.parser
reloading plugin AlignTab.table
reloading plugin AlignTab.wclen
reloading plugin AutoHotkey.ahkexec
reloading plugin Bracketeer.bracketeer
reloading plugin BufferScroll.BufferScroll
reloading plugin ChangeList.changelist
reloading plugin ChannelRepositoryTools.ordereddict
reloading plugin ChannelRepositoryTools.tests
reloading plugin ChannelRepositoryTools.upgrade
reloading plugin ColorSchemeSelector.ColorSchemeSelector
reloading plugin Colorsublime.colorsublime-plugin
reloading plugin Compare Side-By-Side.sbs_compare
reloading plugin Default File Type.default_file_type
reloading plugin EasyMotion.easy_motion
reloading plugin Expand Selection to Quotes.ExpandSelectionToQuotes
reloading plugin Expand Selection to Whitespace.ExpandSelectionToWhitespace
reloading plugin ExpandRegion.__init__
reloading plugin ExpandRegion.expand_region_handler
reloading plugin ExpandRegion.expand_to_indent
reloading plugin ExpandRegion.expand_to_line
reloading plugin ExpandRegion.expand_to_quotes
reloading plugin ExpandRegion.expand_to_regex_set
reloading plugin ExpandRegion.expand_to_semantic_unit
reloading plugin ExpandRegion.expand_to_subword
reloading plugin ExpandRegion.expand_to_symbols
reloading plugin ExpandRegion.expand_to_word
reloading plugin ExpandRegion.expand_to_word_with_dots
reloading plugin ExpandRegion.expand_to_xml_node
reloading plugin ExpandRegion.ExpandRegion
reloading plugin ExpandRegion.html
reloading plugin ExpandRegion.javascript
reloading plugin ExpandRegion.latex
reloading plugin ExpandRegion.python
reloading plugin ExpandRegion.utils
reloading plugin Feature Presentation.FeaturePresentation
reloading plugin File History.file_history
reloading plugin FileBrowser.common
reloading plugin FileBrowser.dired
reloading plugin FileBrowser.dired_file_operations
reloading plugin FileBrowser.dired_misc
reloading plugin FileBrowser.fixpaths
reloading plugin FileBrowser.jumping
reloading plugin FileBrowser.prompt
reloading plugin FileBrowser.show
reloading plugin FindKeyConflicts.find_key_conflicts
reloading plugin GhostText.FocusSublimeWindow
reloading plugin GhostText.GhostText
reloading plugin Gist.gist
reloading plugin Gist.helpers
reloading plugin Gist.request
reloading plugin Gist.settings
reloading plugin GitGutter.git_gutter
reloading plugin GitGutter.git_gutter_change
reloading plugin GitGutter.git_gutter_compare
reloading plugin GitGutter.git_gutter_events
reloading plugin GitGutter.git_gutter_handler
reloading plugin GitGutter.git_helper
reloading plugin GitGutter.view_collection
reloading plugin HighlightWords.HighlightWords
reloading plugin Horizontal Scroll.scroll_width
reloading plugin LoremIpsum.LoremIpsum
reloading plugin Macroptimize.Macroptimize
reloading plugin Markdown Preview.helper
reloading plugin Markdown Preview.markdown_settings
reloading plugin Markdown Preview.markdown_wrapper
reloading plugin Markdown Preview.MarkdownPreview
reloading plugin Nettuts+ Fetch.Fetch
reloading plugin Open URL.example
reloading plugin Open URL.open_url
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin PackageDev.build_sys_dev
reloading plugin PackageDev.commands_file_dev
reloading plugin PackageDev.completions_dev
reloading plugin PackageDev.file_conversion
reloading plugin PackageDev.settings_dev
reloading plugin PackageDev.snippet_dev
reloading plugin PackageDev.syntax_def_dev
reloading plugin PackageResourceViewer.package_resource_viewer
reloading plugin PackageResourceViewer.package_resources
reloading plugin Processing.Processing
reloading plugin Project Manager.closewindow
reloading plugin Project Manager.pm
reloading plugin RegReplace.rr_notify
reloading plugin RegReplace.rr_plugin
reloading plugin RegReplace.rr_replacer
reloading plugin RegReplace.rr_sequencer
reloading plugin ScopeHunter.scope_hunter
reloading plugin ScopeHunter.scope_hunter_notify
reloading plugin SelectUntil.edit
reloading plugin SelectUntil.select-until
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SublimeLinter-javac.linter
SublimeLinter: javac linter loaded 
reloading plugin SublimeLinter-pylint.linter
SublimeLinter: pylint linter loaded 
reloading plugin SuperNavigator.SuperNavigator
reloading plugin Surround.Surround
reloading plugin Table Editor.table_base
reloading plugin Table Editor.table_border_syntax
reloading plugin Table Editor.table_emacs_org_mode_syntax
reloading plugin Table Editor.table_lib
reloading plugin Table Editor.table_lib_test
reloading plugin Table Editor.table_line_parser
reloading plugin Table Editor.table_multi_markdown_syntax
reloading plugin Table Editor.table_pandoc_syntax
reloading plugin Table Editor.table_plugin
reloading plugin Table Editor.table_plugin_test
reloading plugin Table Editor.table_re_structured_text_syntax
reloading plugin Table Editor.table_simple_syntax
reloading plugin Table Editor.table_textile_syntax
reloading plugin Table Editor.widechar_support
reloading plugin TabsExtra.tab_menu
reloading plugin TabsExtra.tab_sort_helper
reloading plugin TabsExtra.tabs_extra
reloading plugin Text Pastry.text_pastry
reloading plugin Text Pastry.text_pastry_addons
reloading plugin Text Pastry.text_pastry_selection
reloading plugin Themr.themr
reloading plugin To Hastebin.ToHasteBin
reloading plugin Trimmer.Trimmer
reloading plugin _DEV_DevTools.DevTools
reloading plugin _DEV_Test.TEST
reloading plugin _MOD_WordHighlight.word_highlight
reloading plugin _PRE__AlignTabQuickPanel.AlignTabQuickPanel
reloading plugin _UTILS._Globals
reloading plugin _UTILS._LOAD_UTILS
reloading plugin _UTILS.Comments
reloading plugin _UTILS.Edit
reloading plugin _UTILS.Indentation
reloading plugin _UTILS.PopUp
reloading plugin _UTILS.RegEx
reloading plugin _UTILS.Region
reloading plugin _UTILS.Settings
reloading plugin _UTILS.Sort
reloading plugin _UTILS.Strings
reloading plugin _UTILS.Verify
reloading plugin _UTILS.ViewData
reloading plugin ASCII Decorator.ASCII-Decorator
Couldn't load pkg_resources
reloading plugin ASCII Decorator.subcomments
reloading plugin ASCII Decorator.subfiglet
reloading plugin AutoHotkey.ahkgotodocs
reloading plugin AutoHotkey.ahkpopuphelp
reloading plugin AutoHotkey.utils
reloading plugin CoolFormat.Coolformat
reloading plugin CustomCompletions.completions DEMO
reloading plugin CustomCompletions.CustomCompletions
reloading plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin Parkour.Parkour [ ToDo ]
reloading plugin Parkour.Parkour
reloading plugin Parkour_v1.module_loader
reloading plugin Parkour_v1.Parkour_v1
reloading plugin Selection Evaluator.mathevaluator
reloading plugin SelectionTools.SelectionTools
plugins loaded





[FileHistory] Reloading the settings file "FileHistory.sublime-settings".
GhostText is starting now…
Setting on request handler
HTTP Start





Package Control: Installing 1 missing dependency
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3103
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Sun, 27 Mar 2016 20:00:00 GMT
    expire date: Sun, 27 Nov 2016 18:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: "57008286-3d1a3"
  If-Modified-Since: Sun, 03 Apr 2016 02:40:06 GMT
  Accept-Encoding: bzip2,gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sun, 03 Apr 2016 02:59:08 GMT
  Content-Type: application/json
  Content-Length: 249563
  Last-Modified: Sun, 03 Apr 2016 02:50:07 GMT
  Connection: keep-alive
  ETag: "570084df-3cedb"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
  Accept-Ranges: bytes
Package Control: Caching https://packagecontrol.io/channel_v3.json in C:\Users\Fico\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Package Control: Download Debug
  URL: https://codeload.github.com/kkga/spacegray/zip/1.2.0
  Timeout: 30
  Resolved IP: 192.30.252.162
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, California, San Francisco, "GitHub, Inc.", *.github.com
    issuer: US, DigiCert Inc, www.digicert.com, DigiCert SHA2 High Assurance Server CA
    common name: *.github.com
    issue date: Mon, 07 Apr 2014 20:00:00 GMT
    expire date: Wed, 12 Apr 2017 08:00:00 GMT
Package Control: WinINet HTTPS Debug Write
  GET /kkga/spacegray/zip/1.2.0 HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: codeload.github.com
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: bzip2,gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Content-Length: 827061
  Access-Control-Allow-Origin: https://render.githubusercontent.com
  Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'
  X-XSS-Protection: 1; mode=block
  X-Frame-Options: deny
  X-Content-Type-Options: nosniff
  Strict-Transport-Security: max-age=31536000
  Vary: Authorization,Accept-Encoding
  ETag: "d60053466a9647caee321962acc03b969b9433e5"
  Content-Type: application/zip
  Content-Disposition: attachment; filename=spacegray-1.2.0.zip
  Date: Sun, 03 Apr 2016 02:59:10 GMT
  X-GitHub-Request-Id: 61649D27:7EEA:B6D2DD:570086FE
Package Control: Skipping cache since the response is a zip file
Package Control: Download Debug
  URL: https://packagecontrol.io/submit?sublime_platform=windows&sublime_version=3103&package_control_version=3.1.2&old_version=1.2.0&package=Theme+-+Spacegray&operation=upgrade&version=1.2.0
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet HTTPS Debug General
  Re-using connection to packagecontrol.io on port 443 for request #1
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug Write
  GET /submit?sublime_platform=windows&sublime_version=3103&package_control_version=3.1.2&old_version=1.2.0&package=Theme+-+Spacegray&operation=upgrade&version=1.2.0 HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: bzip2,gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sun, 03 Apr 2016 02:59:11 GMT
  Content-Type: application/json
  Transfer-Encoding: chunked
  Connection: keep-alive
  Vary: Accept-Encoding
  X-App-Version: 1.3.3
  Strict-Transport-Security: max-age=15552000
  Content-Encoding: gzip
Exception in thread Thread-5:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "package_control.automatic_upgrader in C:\Users\Fico\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 113, in run
  File "package_control.automatic_upgrader in C:\Users\Fico\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 145, in install_missing
  File "package_control.package_manager in C:\Users\Fico\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 1317, in install_package
KeyError: 'load_order'





Package Control: WinINet HTTPS Debug General
  Closing connection to codeload.github.com on port 443 after 1 request
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 2 requests
```
&nbsp;

&nbsp;
Also, sometimes when I open the `console`, this message is printed:
```
reloading Packages/User/Fetch.sublime-settings
```

&nbsp;
I do have a custom `console` launching function, but there's nothing out of the ordinary in it:

```
	def initialize_Console ( window, LOG_ENABLED ):

		newLine_Count = 30

		if LOG_ENABLED == True:
			sublime.log_commands ( True )
			sublime.log_input    ( True )
			window.run_command ( "show_panel", { "panel": "console", "toggle": True } )
			print ( "\n" * newLine_Count )

		elif LOG_ENABLED == False:
			sublime.log_commands ( False )
			sublime.log_input    ( False )
			window.run_command ( "show_panel", { "panel": "console", "toggle": True } )
			print ( "\n" * newLine_Count )
```

**Comment by wbond** [comment]
One of your packages is trying to specify `Theme - Spacegray` as a dependency, which it is definitely not.

I'll need to add some checks to ensure that such things don't happen in the future.

**Comment by wbond** [comment]
This should be resolved by c08f68a0232535dd8b0edfb30df01c93275b2235. It separates the list of dependencies and packages, so if a package developer specified a package as a dependency, it will be listed as not found instead of having an error trying to access fields that don't exist.
